GPU/GPUTracking/TRDTracking/GPUTRDTracker.cxx

```diff
@@ -657,6 +657,22 @@ GPUd() bool GPUTRDTracker_t<TRDTRK, PROP>::FollowProlongation(PROP* prop, TRDTRK
             GPUWarning("Track parameter for track %i, x=%f at chamber %i x=%f in layer %i cannot be retrieved", iTrk, trkWork->getX(), currDet, mR[currDet], iLayer);
           }
         }
+        // check if track crosses padrows
+        if (iDet == 0) {
+          float projZEntry, projYEntry;
+          // Get Z for Entry of Track
+          prop->getPropagatedYZ(trkWork->getX() - 3.f, projYEntry, projZEntry);
+          // Get Padrow number for Exit&Entry and compare. If is not equal mark
+          // as padrow crossing. While simple, this comes with the cost of not
+          // properly handling Tracklets that hit a different Pad but still get
+          // attached to the Track. It is estimated that the probability for
+          // this is low.
+          auto padrowEntry = pad->GetPadRowNumber(projZEntry);
+          auto padrowExit = pad->GetPadRowNumber(trkWork->getX());
+          if (padrowEntry != padrowExit) {
+            trkWork->setIsPadrowCrossing(iLayer);
```

Comment:
Should not this be be validated by the presence of a tracklet in the neighbouring padrow?

Reply:
Yes, when thinking about it this would make more sense. We would probably also want the information about pad row crossing only in case a tracklet is attached. So then you could move this check a bit down to when a tracklet is actually attached to the track. And in that case do in addition the check whether another tracklet is present on the neighboring pad row. Since the tracklets are not sorted within one chamber this probably requires a loop over all tracklet of the given chamber

Reply:
I think the check for tracklets in neighboring padrows is complementary: it would detect if more information is available that should be merged into the already attached tracklet. A similar argument could actually be made for tracklets in the same padrow, especially at low pT and large inclination angle.
The flagging based on tracking only would indicate if some signal was deposited on a neighboring padrow, even it is below the threshold required to generate a tracklet. Even in such a case, the quality of at least one time slice would be compromised, but it would not be detectable by looking for neighboring tracklets.

Reply:
> Are you sure about this? I think low pT tracks spanning over more than 4 pads would rather not be found at all. And for the ones we find (>~700MeV) we should not loose charge information in column direction.

I don't have numbers for this. I remember that a long time ago, we looked into tracklet splitting, and found that a significant number of tracklets (I remember 10-20%) had duplicates, i.e. were reconstructed twice in neighboring channels. But I do not know how how big the differences for the tracklet charges were.

Reply:
> The problem I see when relying purely on the information from the track parameters is that especially for TPC tracks the estimate for the z position might be quite unprecise while eta should be rather well described. This means you can shift the track to some extent along z. And when doing that you can of course always find some position where the track crossed pad rows, but also a position where it did not.

That's a good point. Do you think this would also be a problem for ITS-TPC-TRD tracks? I guess in this case, the timing information should be good enough to fix z?

And to which degree does the TRD fix the z-position of the TPC-TRD tracks? I think we only use 

> Checking if there really were 2 tracklets might be a clearer indication. But that would require a good efficiency for the online tracklets of course. Probably it would be good to check in Run 3 data how often this is the case that for a track two neighboring tracklets are in its vicinity. Could be done both in column and in row direction

It seems that both checks test slightly different things, and both could be useful in different scenarios, e.g. ITS-TPC-TRD tracks without padrow crossings for gain calibration, where you prefer high-quality tracks, and a check for neighboring tracklets for PID, where you are more interested in efficiency.

I would therefore implement both.

Reply:
> ITS-TPC-TRD tracks without padrow crossings for gain calibration, where you prefer high-quality tracks
Do you(@tdietel) want to generally throw away tracks if they have a padrow crossing for gain calibration or would you like to have this layerwise?